### PR TITLE
refactor: Remove util dependency

### PR DESCRIPTION
**Pull request by nbbeeken**
Removes util dependency and restricts rollup bundler to only include Buffer

NODE-2826

### CODE REVIEW

**Review by emadum — APPROVED** [review]
LGTM

File: src/parser/utils.ts [hunk 1]
```diff
@@ -60,3 +61,19 @@ export function isDate(d: unknown): d is Date {
 export function isObjectLike(candidate: unknown): candidate is Record<string, unknown> {
   return typeof candidate === 'object' && candidate !== null;
 }
+
+export function deprecate<T extends Function>(fn: T, message: string): T {
+  if (typeof window === 'undefined' || typeof self === 'undefined') {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    return require('util').deprecate(fn, message);
```
Comment (mbroadst) [inline src/parser/utils.ts:68]: Is there still any benefit to using the Node built-in at this point? I like this runtime agnostic one

**Review by mbroadst — CHANGES_REQUESTED** [review]

**Review by reggi — APPROVED** [review]

File: src/objectid.ts [hunk 2]
```diff
@@ -422,6 +421,6 @@ Object.defineProperty(ObjectId, 'get_inc', {
  * @returns return the 24 character hex string representation.
  * @internal
  */
-Object.defineProperty(ObjectId.prototype, inspect.custom || 'inspect', ObjectId.prototype.toString);
+Object.defineProperty(ObjectId.prototype, 'inspect', ObjectId.prototype.toString);
```
[the remark this answers is not in the document]
Comment (nbbeeken) [reply src/objectid.ts:373]: Good idea, the inspect symbol will always exist here, Symbol.for creates a new symbol if one doesn't exist. So I'll drop the if statement.

File: src/parser/utils.ts (hunk 1 above)
Comment (nbbeeken) [reply src/parser/utils.ts:81]: If we're in nodejs this will handle the `--throw-deprecation`, `--no-deprecation`, `--trace-deprecation` flags.

**Review by mbroadst — APPROVED** [review]
LGTM

**Pull request merged**